src/__tests__/new-rule-yaml.test.ts

```
@@ -0,0 +1,1147 @@
+      mockFs.existsSync.mockReturnValue(true);
+
+      const newRule = {
+        name: 'formatting-test-rule',
+        description: 'Test rule for formatting preservation',
+        enabled: true,
+        conditions: ['format-condition']
+      };
+
+      addNewRule.mockImplementation((filePath: string, rule: any) => {
+        // Mock preserving formatting
+        mockFs.writeFileSync(filePath, yamlWithComments + '\n  - name: formatting-test-rule');
```

Biome (1.9.4)

[error] Template literals are preferred over string concatenation.

Unsafe fix: Use a template literal.


(lint/style/useTemplate)